src/cljs/athens/views/blocks.cljs

```diff
@@ -221,7 +223,18 @@
 (defn on-change
   [value uid]
   ;; (prn "ONCHANGE" value)
-  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]]))
+  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]])
+  ;; automatically add non-existent pages
+  ;; TODO: delete pages that are no longer connected to anything else
+  (parse/transform {:page-link (fn [& title]
+                                 (if (nil? (db/search-exact-node-title (apply + title)))
+                                   (let [uid (gen-block-uid)]
+                                     (dispatch [:transact [{:node/title     (apply + title)
+                                                            :block/uid      (str uid)
+                                                            :edit/time      (now-ts)
+                                                            :create/time    (now-ts)}]]))
+                                   nil)
```

Comment:
`if` statement with a `nil` for the else clause should just use `when`

Reply:
Fixed